x/posts/keeper/polls.go

```diff
@@ -62,6 +63,27 @@ func (k Keeper) Tally(ctx sdk.Context, subspaceID uint64, postID uint64, pollID
 	return types.NewPollTallyResults(tallyResults)
 }
 
+// EndPoll tallies the poll then update it inside storage
+func (k Keeper) EndPoll(ctx sdk.Context, poll types.Attachment) {
+	// Compute the poll results
+	results := k.Tally(ctx, poll.SubspaceID, poll.PostID, poll.ID)
+
+	// Update the content with the results
+	content := poll.Content.GetCachedValue().(*types.Poll)
+	content.FinalTallyResults = results
+
+	contentAny, err := codectypes.NewAnyWithValue(content)
+	if err != nil {
+		panic(err)
+	}
+	poll.Content = contentAny
+
+	k.SaveAttachment(ctx, poll)
+	k.RemoveFromActivePollQueue(ctx, poll)
+
+	k.AfterPollVotingPeriodEnded(ctx, poll.SubspaceID, poll.PostID, poll.ID)
+}
```

Comment:
To allow to test `AfterPollVotingPeriodEnded` inside the keeper, it is moved here from abci.

Reply:
Why was this decided? I believe we can call it safely in every occasion: if we are removing a user answer from the store, we should notify it anyway. It will be the listener's job to decide whether to handle it or not. And since the tally emits a custom event, it should be easy enough to distinguish the two cases if the listener wants